impl/client/src/main/javascript/web/pentaho/type/Context.js

```diff
@@ -16,40 +16,278 @@
 define([
   "require",
   "module",
-  "../service",
   "../typeInfo",
   "../i18n!types",
-  "./standard",
+  "./InstanceConstructionContext",
   "./SpecificationContext",
   "./SpecificationScope",
+  "./InstancesContainer",
   "../environment",
   "../service!pentaho.config.IService?single",
   "./changes/Transaction",
   "./changes/TransactionScope",
   "./changes/CommittedScope",
   "../lang/Base",
+  "./util",
   "../util/promise",
   "../util/arg",
   "../util/error",
   "../util/object",
-  "../util/fun"
-], function(localRequire, module, service, typeInfo, bundle, standard, SpecificationContext, SpecificationScope,
-    mainPlatformEnv, configurationService,
+  "../util/fun",
+  "../debug",
+  "../debug/Levels",
+  "../util/logger",
+  "./standard" // so that r.js sees otherwise invisible dependencies.
```

Comment:
Can't this be handled in r.js configuration?

Reply:
In theory yes. In practice, it would be much less practical. It is Analyzer that is doing the r.js build. I just think that it doesn't kill us to, in some cases, *also* declare the dependencies in the AMD module directly.